src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -63,13 +64,13 @@ internal sealed class RestoreOperationLogger : ILogger
 
         public async Task StartAsync(
             RestoreOperationSource operationSource,
-            ErrorListProvider errorListProvider,
+            Lazy<ErrorListTableDataSource> errorListDataSource,
```

Comment:
Consider passing it as a constructor argument. This'd let MEF resolve the dependency and would simplify `StartAsync` contract.